roles/mirror_ocp_release/tasks/installer.yml

```diff
@@ -1,32 +1,42 @@
 ---
 - name: "Check if installer has been extracted"
   ansible.builtin.stat:
-    path: "{{ mor_cache_dir }}/{{ mor_version }}/openshift-baremetal-install"
+    path: "{{ mor_cache_dir }}/{{ mor_version }}/{{ item }}"
     get_checksum: false
   register: target
+  loop:
+    - openshift-baremetal-install
+    - openshift-install
   when:
-    - not mor_force  # we don't care to stat files if we're forcing
+    - not mor_force | bool # we don't care to stat files if we're forcing
 
 - name: "Extract installer command from release image"
+  when:
+    - ((mor_force | bool) or
+      (not target | json_query('results[*].stat.exists') | unique == [true]))
   ansible.builtin.command: >
     {{ mor_oc }} adm release extract
     --registry-config={{ mor_auths_file }}
-    --command=openshift-baremetal-install
+    --command={{ item }}
     --from {{ mor_pull_url }}
     --to "{{ mor_cache_dir }}/{{ mor_version }}"
   register: extract_cmd
   until: extract_cmd is succeeded
   retries: 9
   delay: 10
-  when:
-    - mor_force or not target.stat.exists
+  loop:
+    - openshift-baremetal-install
+    - openshift-install
```

Comment:
Do we need to loop both commands here ? or could {{ cmd }} be used ?

Reply:
I followed your suggestion and used a similar approach as in the installer role